### PR TITLE
Upgrade to kubernetes 1.21 / libraries 0.21.0

### PR DESCRIPTION
**Pull request by sbose78**
**What this PR does / why we need it**:
Use the latest and greatest client libraries for Kubernetes APIs

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Comment by sbose78** [comment]
Would this need updates to the doc https://helm.sh/docs/topics/version_skew/ ?

File: pkg/action/lazyclient.go [hunk 1]
```diff
@@ -116,6 +117,13 @@ func (s *secretClient) Patch(ctx context.Context, name string, pt types.PatchTyp
 	return s.client.CoreV1().Secrets(s.namespace).Patch(ctx, name, pt, data, opts, subresources...)
 }
 
+func (s *secretClient) Apply(ctx context.Context, secretConfiguration *applycorev1.SecretApplyConfiguration, opts metav1.ApplyOptions) (*v1.Secret, error) {
```
Comment (bacongobbler) [inline pkg/action/lazyclient.go:120]: I'm assuming these are new functions introduced to an interface somewhere. Do you have a link handy to the changes so we can investigate further?

**Comment by bacongobbler** [comment]
Looks good. Just want to dive deeper into the changes to see if there's something we're missing here.

File: pkg/action/lazyclient.go (hunk 1 above)
Comment (sbose78) [reply pkg/action/lazyclient.go:120]: https://github.com/helm/helm/blob/257a427866883a9fb745be40800ffe8e9ceda82b/pkg/action/lazyclient.go#L58

This line caught the changes :)

File: pkg/action/lazyclient.go [hunk 2]
```diff
@@ -60,7 +61,7 @@ func newSecretClient(lc *lazyClient) *secretClient {
 	return &secretClient{lazyClient: lc}
 }
 
-func (s *secretClient) Create(ctx context.Context, secret *v1.Secret, opts metav1.CreateOptions) (*v1.Secret, error) {
+func (s *secretClient) Create(ctx context.Context, secret *v1.Secret, opts metav1.CreateOptions) (result *v1.Secret, err error) {
```
Comment (mattfarina) [inline pkg/action/lazyclient.go:64]: Why was the return signature changed to `(result *v1.Secret, err error)`?

**Review by mattfarina — COMMENTED** [review]
Note, k8s 1.21.1 was just released so an update will be coming to the libs soon. They aren't available as of this comment. Just wanted to give a heads up.

**Review by adamreese — APPROVED** [review]

**Review by mattfarina — APPROVED** [review]

File: pkg/action/lazyclient.go (hunk 2 above)
Comment (sbose78) [reply pkg/action/lazyclient.go:64]: Hi Matt,
This was changed to align with the updated signature in https://pkg.go.dev/k8s.io/client-go/kubernetes/typed/core/v1#SecretInterface

**Pull request merged**